scripts/perf.js

```diff
@@ -0,0 +1,146 @@
+const path = require('path');
+const fs = require('fs');
+const rollup = require('../dist/rollup.js');
+const chalk = require('chalk');
+const { loadPerfConfig, targetDir } = require('./load-perf-config');
+
+const perfFile = path.resolve(targetDir, 'rollup.perf.json');
+
+let numberOfRunsToAverage = 6;
+let numberOfDiscardedResults = 1;
+if (process.argv.length >= 3) {
+	numberOfRunsToAverage = Number.parseInt(process.argv[2]);
+	if (process.argv.length >= 4) {
+		numberOfDiscardedResults = Number.parseInt(process.argv[3]);
+	}
+}
+if (!(numberOfDiscardedResults >= 0) || !(numberOfDiscardedResults < numberOfRunsToAverage)) {
+	console.error(
+		`Invalid parameters: runs = ${numberOfRunsToAverage}, discarded = ${numberOfDiscardedResults}.\n` +
+			'Usage: "npm run perf [<number of runs> [<number of discarded results>]]"\n' +
+			'where 0 <= <number of discarded results> < <number of runs>'
+	);
+	process.exit(1);
+}
+console.info(
+	chalk.bold(
+		`Calculating the average of ${chalk.cyan(numberOfRunsToAverage)} runs discarding ${chalk.cyan(
+			numberOfDiscardedResults
+		)} outliers.\n`
+	) + 'Run "npm run perf <number of runs> <number of discarded results>" to change that.'
+);
+
+function getAverage(times) {
+	const initialAverage = times.reduce((sum, time) => sum + time, 0) / numberOfRunsToAverage;
+	return (
+		times
+			.map(time => ({ time, deviation: Math.abs(time - initialAverage) }))
+			.sort(({ deviation: dev1 }, { deviation: dev2 }) => dev2 - dev1)
+			.slice(numberOfDiscardedResults)
+			.reduce((sum, { time }) => sum + time, 0) /
+		(numberOfRunsToAverage - numberOfDiscardedResults)
+	);
+}
+
+async function getAverageTimings(config) {
+	await buildAndGetTimings(config);
+	console.info('Completed initial run (Discarded).');
+	const timings = await buildAndGetTimings(config);
+	console.info('Completed run 1.');
```

Comment:
It could be useful to have some basic timing information as these show for quick perf testing when one doesn't want to wait for all the runs to complete (thing eg ping).